### PR TITLE
Add expose

### PR DESCRIPTION
**Pull request by huwshimi**
Add the expose view to the inspector.

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- add a service to the canvas
- check that the inspector does not have an Expose item
- place the unit on a machine
- deploy
- check that the inspector now has an Expose item
- check that the value next to Expose is "Off"
- click on Expose
- click on the toggle to expose the service
- a list of units should appear
- click a unit to view the unit
- click the header to go back
- you should be taken back to the expose view
- click the header to go back
- check that the value next to Expose is "On"
- deploy
- check that the service still has Expose "On"

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3256/
Test PASSed.

**Comment by hatched** [comment]
I see you made a few modifications to the boolean-config component, maybe it should be renamed something like 'toggle' in a follow-up. Thoughts?

File: jujugui/static/gui/src/app/components/inspector-expose/inspector-expose.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+                key={toggle.key}
+                ref={toggle.key}
+                option={toggle}
+                onChange={this._handleExposeChange}
+                label="Expose service"
+                config={this.props.service.get('exposed')} />
+            </div>
+            <p className="inspector-expose__warning">
+              Exposing this service will make it publicly accessible from
+              the web
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/inspector-expose/inspector-expose.js]: I'm not sure this message is technically valid as all it does is open the ports.

**Comment by hatched** [comment]
:+1: This is so awesome, however I'm a little concerned that the expose button isn't visible until you've placed units on a machine I'm assuming this was direction from design?

QA Issue:
When removing a unit it redirects back to the unit list instead of the expose pane.

**Comment by huwshimi** [comment]
@hatched yep, we could rename the component at some stage, no big deal. As discussed we'll leave the expose button hidden pre-commit to match the existing functionality.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3268/
Test PASSed.

**Comment by hatched** [comment]
QA :ok: Thanks for adding this!

**Comment by huwshimi** [comment]
Thanks @hatched :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Build failed: Attempt to land pull request failed
        build url: http://ci.jujugui.org:8080/job/juju-gui-merge/1114

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**